### PR TITLE
Advanced auth front

### PR DESCRIPTION
**Pull request by AnvarjonShavqiyev**
Integrated JWT authentication in the React app.
Created login and registration forms.
Stored the JWT token securely in the browser with localstorage.
Used the token to access protected endpoints.
 
photo_2024-07-05_17-48-00
photo_2024-07-05_17-48-27
photo_2024-07-05_17-48-36
photo_2024-07-05_17-48-47
photo_2024-07-05_17-59-17
photo_2024-07-05_17-59-29 (2)
photo_2024-07-05_17-59-29

### CODE REVIEW

**Review by EugeneSinitsa — CHANGES_REQUESTED** [review]

File: my-react-app/src/pages/Auth/signin/Signin.tsx [hunk 1]
```diff
@@ -2,13 +2,13 @@ import { Form, Input, Button } from 'antd';
...
-import { signInUser } from '../../../redux/features/AuthSlice';
-import { User } from '../../../types';
+import { signUpUser } from '../../../redux/features/AuthSlice';
 
 const Signin = () => {
   const dispatch = useDispatch<AppDispatch>()
-  const onFinish = (values:any) => {
-    dispatch(signInUser({ email: values.email }));
+  const onFinish = (values: { email: string; password: string }) => {
+    console.log('Signin values:', values);
```
[the remark this answers is not in the document]
Comment (AnvarjonShavqiyev) [reply my-react-app/src/pages/Auth/signin/Signin.tsx]: fixed

File: my-react-app/src/redux/features/AuthSlice.ts [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
 
 export const createUser = createAsyncThunk<User, { newUser: any }, { rejectValue: string }>(
   'auth/createUser',
   async ({ newUser }, thunkAPI) => {
     try {
-      const response: AxiosResponse<User> = await instance.post("/users", newUser);
+      const response: AxiosResponse<User> = await instance.post("/auth/signup", newUser);
       return response.data;
     } catch (error: any) {
       console.log('Error creating user:', error);
```
[the remark this answers is not in the document]
Comment (AnvarjonShavqiyev) [reply my-react-app/src/redux/features/AuthSlice.ts]: fixed

File: my-react-app/src/redux/features/UsersSlice.ts [hunk 3]
```diff
@@ -24,7 +24,12 @@ export const getAllUsers = createAsyncThunk<User[], void, { rejectValue: string
   'user/getAllUsers',
   async (_, thunkAPI) => {
     try {
-      const response: AxiosResponse<User[]> = await instance.get("/users");
+      const token = localStorage.getItem('token')
+      const response: AxiosResponse<User[]> = await instance.get("/users",{
+        headers:{
+          Authorization: `Bearer ${token}`
```
[the remark this answers is not in the document]
Comment (AnvarjonShavqiyev) [reply my-react-app/src/redux/features/UsersSlice.ts:30]: Yeah, that is better way, i got it.

**Comment by esnts** [comment]
merge it

**Pull request merged**